server/src/main/java/org/opensearch/wlm/cancellation/DefaultTaskSelectionStrategy.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm.cancellation;
+
+import org.opensearch.wlm.QueryGroupTask;
+import org.opensearch.wlm.ResourceType;
+import org.opensearch.wlm.tracker.TaskResourceUsageCalculator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static org.opensearch.wlm.cancellation.DefaultTaskCancellation.MIN_VALUE;
+
+/**
+ * Represents an abstract task selection strategy.
+ * This class implements the DefaultTaskSelectionStrategy interface and provides a method to select tasks for cancellation based on a sorting condition.
+ * The specific sorting condition depends on the implementation.
+ */
+public class DefaultTaskSelectionStrategy {
+
+    private final Supplier<Long> nanoTimeSupplier;
+
+    public DefaultTaskSelectionStrategy() {
+        this(System::nanoTime);
+    }
+
+    public DefaultTaskSelectionStrategy(Supplier<Long> nanoTimeSupplier) {
+        this.nanoTimeSupplier = nanoTimeSupplier;
+    }
+
+    /**
+     * Returns a comparator that defines the sorting condition for tasks.
+     * This is the default implementation since the longest running tasks are the ones that consume the most resources.
+     *
+     * @return The comparator
+     */
+    public Comparator<QueryGroupTask> sortingCondition() {
+        return Comparator.comparingLong(QueryGroupTask::getStartTime);
+    }
```

Comment:
Since we are averaging the utilization on the basis of task's running time, not sure if the assumption of `longest running tasks consuming most resources` really holds true?

Reply:
I agree the comment is bit misleading. I will change it